src/script/builder/tir/stmt.cc

```diff
@@ -0,0 +1,265 @@
+}
+
+AllocateFrame Allocate_(Array<PrimExpr> extents, DataType dtype, String storage_scope_str,
+                        PrimExpr condition, Optional<Map<String, ObjectRef>> annotations) {
+  ObjectPtr<AllocateFrameNode> n = make_object<AllocateFrameNode>();
+  n->extents = extents;
+  n->dtype = dtype;
+  n->storage_scope_str = storage_scope_str;
+  n->condition = condition->dtype.is_bool() ? condition : tvm::cast(DataType::Bool(), condition);
+  n->annotations = annotations.value_or(Map<String, ObjectRef>());
+  n->buffer = DeclBuffer(extents, dtype, "", NullOpt, {}, PrimExpr(), storage_scope_str, 0, 0,
+                         "default", {}, Span());
+  return AllocateFrame(n);
+}
+
+AllocateConstFrame AllocateConst_(tvm::runtime::NDArray data, DataType dtype,
+                                  Array<PrimExpr> extents) {
+  ObjectPtr<AllocateConstFrameNode> n = make_object<AllocateConstFrameNode>();
+  n->dtype = dtype;
+  n->extents = extents;
+  n->data = data;
+  n->buffer =
+      DeclBuffer(extents, dtype, "", NullOpt, {}, PrimExpr(), "", 0, 0, "default", {}, Span());
+  return AllocateConstFrame(n);
+}
+
+LaunchThreadFrame LaunchThread(tvm::tir::IterVar env_var, PrimExpr extent) {
+  ObjectPtr<LaunchThreadFrameNode> n = make_object<LaunchThreadFrameNode>();
+  n->env_var =
+      tvm::tir::IterVar(Range(0, extent), env_var->var, env_var->iter_type, env_var->thread_tag);
+  n->extent = extent;
+  n->attr_key = env_var->thread_tag == "vthread" ? "virtual_thread" : "thread_extent";
+  return LaunchThreadFrame(n);
+}
+
+RealizeFrame Realize(tvm::tir::BufferRegion buffer_slice, String storage_scope_str,
+                     PrimExpr condition) {
+  ObjectPtr<RealizeFrameNode> n = make_object<RealizeFrameNode>();
+  n->buffer_slice = buffer_slice;
+  n->storage_scope_str = storage_scope_str;
+  n->condition = condition;
+  return RealizeFrame(n);
+}
+
+AttrFrame Attr(ObjectRef node, String attr_key, PrimExpr value) {
+  ObjectPtr<AttrFrameNode> n = make_object<AttrFrameNode>();
+  n->node = node;
+  n->attr_key = attr_key;
+  n->value = value;
+  return AttrFrame(n);
+}
+
+WhileFrame While_(PrimExpr condition) {
+  ObjectPtr<WhileFrameNode> n = make_object<WhileFrameNode>();
+  n->condition = condition;
+  return WhileFrame(n);
+}
+
+IfFrame If_(PrimExpr condition) {
+  ObjectPtr<IfFrameNode> n = make_object<IfFrameNode>();
+  n->condition = condition;
+  n->then_stmt.clear();
+  n->else_stmt.clear();
+  return IfFrame(n);
+}
+
+IfFrame FindIfFrame(const String& method) {
+  if (Optional<IfFrame> if_frame = Builder::Current()->GetLastFrame<IfFrame>()) {
+    return if_frame.value();
+  } else {
+    LOG(FATAL) << "ValueError: IfThenElse frame not find. Please ensure '" << method
+               << "' is called under T.if_()";
+  }
+  throw;
+}
+
+ThenFrame Then_() {
+  IfFrame frame = FindIfFrame("T.then_");
+  if (!frame->then_stmt.empty()) {
+    LOG(FATAL) << "Duplicate then branch declaration, previous one is " << frame->then_stmt;
```

Comment:
Let's always prefix the error msg with an error type

```suggestion
    LOG(FATAL) << "ValueError: Duplicate then branch declaration, previous one is " << frame->then_stmt;
```